Sources/Features/Breadcrumb/BacktraceBreadcrumbFileHelper.swift

```diff
@@ -5,38 +5,38 @@ enum BacktraceBreadcrumbFileHelperError: Error {
     case invalidFormat
 }
 
-@objc public class BacktraceBreadcrumbFileHelper: NSObject {
+@objc class BacktraceBreadcrumbFileHelper: NSObject {
 
     /*
      The underlying library CASQueueFile assigns a minimum of 4k (filled with zeroes).
      Since we know that space will be allocated (and uploaded) anyways, set it as the minimum.
      */
     private static let minimumQueueFileSizeBytes = 4096
 
-    /* We cap the size of an individual breadcrumb at 4k, for performance reasons. */
-    private static let maximumBreadcrumbSize = 4096
-
+    private let maximumIndividualBreadcrumbSize: Int
     private let maxQueueFileSizeBytes: Int
-    private let breadcrumbLogDirectory: String
-
     private let queue: CASQueueFile
 
-    public init(_ breadcrumbLogDirectory: String, maxQueueFileSizeBytes: Int) throws {
+    /** CASQueueFile is not thread safe, so all interactions with it should be done synchronously through this DispathQueue */
+    private let dispatchQueue = DispatchQueue(label: "io.backtrace.BacktraceBreadcrumbFileHelper@\(UUID().uuidString)")
+
+    public init(_ breadcrumbSettings: BacktraceBreadcrumbSettings) throws {
         do {
-            self.queue = try CASQueueFile.init(path: breadcrumbLogDirectory)
+            self.queue = try CASQueueFile.init(path: breadcrumbSettings.getBreadcrumbLogPath().path)
         } catch {
             BacktraceLogger.error("\(error.localizedDescription) \nWhen enabling breadcrumbs")
             throw error
```

Comment:
we shouldn't rethrow an error here. Why? It might cause an infinity loop of crashes. How? when the directory is not available on the startup or something unexpected happened to the directory, then the code will rethrow an exception. What can we do instead? Wouldn't it be better IF we return boolean from the init method and depends on the result, continue initialization or disable breadcrumbs?

Reply:
Agreed. I think it may be better to return nil here (you can do that in Swift).

Reply:
Actually, I think it's less graceful that way. Swift DOES allow you to return nil, but then we get in the situation the nil needs to be handled in the invoking class (LogManager). I think the change should be to not catch the error. Then all the problems constructing the dependencies are capture in BacktraceBreadcrumbs and they are left turned off.

Does that work? I'll push that change now.

Reply:
Done